### PR TITLE
CCP-171 Changes for wrapping concept while editing

### PR DESCRIPTION
**Pull request by mkarthik90**

### CODE REVIEW

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java [hunk 1]
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
...
+                        HttpStatus.SERVICE_UNAVAILABLE);
+            }
+
+            foundConcepts = new CopyOnWriteArrayList<>(wrapperCreator.createWrappers(found));
+        }
+        
+        // Remove CCP concepts. 
+        for(ConceptEntryWrapper wrapper : foundConcepts) {
+            System.out.println(wrapper.getEntry().getId());
+            if (wrapper.getEntry().getId().startsWith(Constants.CONCEPT_PREFIX)) {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: Isn't there a method to test concept ids for what type they are?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java [hunk 2]
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
...
+                return new ResponseEntity<Object>("Indexer is running. Try again after sometime.",
+                        HttpStatus.SERVICE_UNAVAILABLE);
+            }
+
+            foundConcepts = new CopyOnWriteArrayList<>(wrapperCreator.createWrappers(found));
+        }
+        
+        // Remove CCP concepts. 
+        for(ConceptEntryWrapper wrapper : foundConcepts) {
+            System.out.println(wrapper.getEntry().getId());
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: ?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java [hunk 3]
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
...
+            if (wrapper.getEntry().getId().startsWith(Constants.CONCEPT_PREFIX)) {
+                foundConcepts.remove(wrapper);
+            }
+        }
+        
+        if (foundConcepts == null || foundConcepts.isEmpty()) {
+            return new ResponseEntity<Object>(HttpStatus.NOT_FOUND);
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: should just return empty results. nothing special here, just no results.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java [hunk 4]
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
         return new ResponseEntity<String>(buildJSON(synonymList, false, true), HttpStatus.OK);
     }
 
+    @RequestMapping(method = RequestMethod.GET, value = "conceptEdit/conceptSearch")
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: make that just .../search

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java [hunk 5]
```diff
@@ -258,6 +267,47 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
...
+    public @ResponseBody ResponseEntity<Object> searchConcept(@RequestParam("concept") String concept,
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: Doesn't this method do the same thing as the method that is called when creating a new concept? Can't you reuse, or combine these two?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java (hunk 1 above)
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: @jdamerow  Yes there is a class called ConceptTypesService but that class has been added as part of CCP-163. May be i will create this PR once that is merged

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java (hunk 5 above)
Comment (mkarthik90) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: @jdamerow  Yes I was thinking the same. While creating the concept we are returning the model object but here we are returning the response entity. May be I can combine the logic of searching a concept. While create a new concept we also get the concept list and concept types details, Do we need the same details here too?

**Comment by mkarthik90** [comment]
@jdamerow  Addressed review comments.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java (hunk 5 above)
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: Yes

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java [hunk 6]
```diff
@@ -258,6 +272,43 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
...
+            } catch (IndexerRunningException ie) {
+                return new ResponseEntity<Object>("Indexer is running. Try again after sometime.",
+                        HttpStatus.SERVICE_UNAVAILABLE);
+            }
+
+            foundConcepts = new CopyOnWriteArrayList<>(wrapperCreator.createWrappers(found));
+        }
+        
+        // Remove CCP concepts. 
+        for(ConceptEntryWrapper wrapper : foundConcepts) {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: If you search for only concepts in Wordnet list (line 292), do you need this here?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java [hunk 7]
```diff
@@ -258,6 +272,43 @@ public ModelAndView confirmEdit(@PathVariable("id") String id, HttpServletReques
         return new ResponseEntity<String>(buildJSON(synonymList, false, true), HttpStatus.OK);
     }
 
+    @RequestMapping(method = RequestMethod.GET, value = "conceptEdit/search")
+    public @ResponseBody ResponseEntity<Object> searchConcept(@RequestParam("concept") String concept,
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: This code is still duplicating existing code, isn't it? Extract duplicate code and reuse it.

Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/web/ConceptEditController.java]: Hm, ya, two lines are the same and the structure... I'll create a new ticket to refactor the wrapper creator. For now let's leave it the way it is.

**Comment by mkarthik90** [comment]
@jdamerow  Addressed review comments. 

I have not merged the logic of searching in ConceptWrapperAddController ConceptEditController. Only the code for fetching the concept entry looks similar. Please let me know if I need to still create a common method.

**Comment by jdamerow** [comment]
ok to test

**Comment by jdamerow** [comment]
Let's make a little button next to each row that lets users add a concept to the wrapped concepts:
image

**Comment by jdamerow** [comment]
Also, the search shows regular concepts as well. Is the lucene search working correctly?

**Comment by mkarthik90** [comment]
Will merge to CCP-176

**Pull request closed without merging**